### PR TITLE
(DOCSP-19772): VSCode export to language

### PR DESCRIPTION
**Pull request by jeff-allen-mongo**
### Description

New VS Code Extension feature that allows exporting playground code to a programming language.

### JIRA

[DOCSP-19772](https://jira.mongodb.org/browse/DOCSP-19772)

### Staging

[Export a Query or Pipeline to Language](https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-vscode/docsworker-xlarge/DOCSP-19772/export-to-language/)

### CODE REVIEW

File: source/includes/steps-open-new-playground.yaml [hunk 1]
```diff
@@ -21,14 +21,13 @@ content: |
...
+   with a few commands.
 
    .. note::
 
-      You can load new Playgrounds without the template by
-      disabling the :guilabel:`Use Default Template For Playground`
-      setting. To learn more about |vsce| settings, see
-      :ref:`vsce-settings`.
+      To load new Playgrounds without the template, disable the
+      :guilabel:`Use Default Template For Playground` setting. To learn
```
Comment (davemungo) [inline source/includes/steps-open-new-playground.yaml:30]: (Throughout) :guilabel:`Use Default Template For Playground` doesn't match the normal rendering of a parameter. It's not the same as for an internal parameter. Updating it across these docs would be too much scope creep, but the rendering seems really inconsistent.

File: source/export-to-language.txt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+You can export query documents and aggregation pipelines from a
```
Comment (davemungo) [inline source/export-to-language.txt:15]: Are query documents and aggregation pipelines the only things you can export? (The shell will take variables as well. Not sure what the drives can bring in.)

Maybe something more general like,

"You can export code developed in  :ref:`playground <vsce-playgrounds>` " ...

**Review by davemungo — COMMENTED** [review]
Jeff - 

Thanks for the updates. Most of these files are includes or steps. It's difficult to see the changes in staging. Next time could you post links to where the includes are visible? 

I made one or two small content suggestions. I don't know what the pre-existing structure is, but if any of these includes/steps could be flattened it would be great.

File: source/export-to-language.txt (hunk 2 above)
Comment (jeff-allen-mongo) [reply source/export-to-language.txt:15]: You can only export query documents and aggregation pipelines.

File: source/export-to-language.txt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+:ref:`playground <vsce-playgrounds>` to a programming language. You can
```
[the remark this answers is not in the document]
Comment (jeff-allen-mongo) [reply source/export-to-language.txt:16]: I'd like to stick with export generally because that is the language used in the UI. I think calling it "translate" might confuse users since they can map what they see in the docs 1:1 to what's in the extension UI.

I will try to work "translate" into this opening sentence.

File: source/export-to-language.txt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- C# 
+- Java 
+- Node
+- Python
+
+Prerequisites
+-------------
+
+You must open a playground that contains a query document or pipeline
+you want to export. These tutorials use the default playground template.
```
[the remark this answers is not in the document]
Comment (jeff-allen-mongo) [reply source/export-to-language.txt:28]: Moved this around a bit.

File: source/includes/blank-playground.rst [hunk 5]
```diff
@@ -1,2 +1,2 @@
-To use this example, **start with a blank MongoDB Playground** by
-clearing the template Playground if it is loaded.
\ No newline at end of file
+To run this example, **start with a blank MongoDB Playground** by
+clearing the template Playground if it is loaded.
```
[the remark this answers is not in the document]
Comment (jeff-allen-mongo) [reply source/includes/blank-playground.rst:2]: I thought I was going to use this include but didn't end up using it for these changes, so edits here are somewhat moot. Apologies for the confusion.

"Clearing the template playground" just means selecting all the code and deleting it.

File: source/includes/fact-export-options.rst [hunk 6]
```diff
@@ -0,0 +1,6 @@
+You can choose whether to include import statements, driver syntax,
```
[the remark this answers is not in the document]
Comment (jeff-allen-mongo) [reply source/includes/fact-export-options.rst:1]: These includes are the beginning of step 3 of both procedures. I'd like to keep them as includes because the copy is the same throughout. If we need to make edits to this copy, it will only need to be done in one place instead of two.

File: source/includes/steps-open-new-playground.yaml (hunk 1 above)
Comment (jeff-allen-mongo) [reply source/includes/steps-open-new-playground.yaml:30]: That is a fair comment, but we have typically used `guilabel` for settings found in a UI, rather than specified in code. That's the distinction we're making.

File: source/includes/fact-export-selection.rst [hunk 7]
```diff
@@ -0,0 +1,6 @@
+a. When you highlighted your code, a light bulb icon appeared. Click
```
Comment (jeff-allen-mongo) [inline source/includes/fact-export-selection.rst:1]: This include appears at the top of step 2 in both procedures.

**Review by jeff-allen-mongo — COMMENTED** [review]
@davemungo , thanks for the review. Please see comments and edits.

Jeff

File: source/includes/fact-export-selection.rst (hunk 7 above)
Comment (jeff-allen-mongo) [reply source/includes/fact-export-selection.rst:1]: Staged here:

- https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-vscode/docsworker-xlarge/DOCSP-19772/export-to-language/#export-your-selection
- https://docs-mongodbcom-staging.corp.mongodb.com/mongodb-vscode/docsworker-xlarge/DOCSP-19772/export-to-language/#export-your-selection-1

Comment (davemungo) [reply source/includes/fact-export-selection.rst:1]: Done

Comment (davemungo) [reply source/includes/fact-export-selection.rst:1]: (I don't see a way to resolve the comment in this side-by-side view. I'll check the conversation view afterwards.

File: source/includes/blank-playground.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (davemungo) [reply source/includes/blank-playground.rst:2]: Done.

File: source/includes/fact-export-options.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (davemungo) [reply source/includes/fact-export-options.rst:1]: Done.

File: source/includes/steps-open-new-playground.yaml (hunk 1 above)
Comment (davemungo) [reply source/includes/steps-open-new-playground.yaml:30]: Done 

Is that our choice or something passed over to us from the design team? We should think about changing it since the current style looks like we've messed up our markup.

File: source/export-to-language.txt (hunk 2 above)
Comment (davemungo) [reply source/export-to-language.txt]: Done

File: source/export-to-language.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (davemungo) [reply source/export-to-language.txt]: Done

File: source/export-to-language.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (davemungo) [reply source/export-to-language.txt]: Done

File: source/includes/fact-export-selection.rst [hunk 8]
```diff
@@ -0,0 +1,6 @@
+a. Click the light bulb icon that appears after you highlight your
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/includes/fact-export-selection.rst]: Done

**Review by mmarcon — APPROVED** [review]
LGTM 🚀

File: source/export-to-language.txt [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+   :depth: 1
+   :class: singlecol
+
+You can export and translate query documents and aggregation pipelines
+from a :ref:`playground <vsce-playgrounds>` into a programming language.
+You can export queries and pipelines to the following languages:
+
+- C# 
+- Java 
+- Node
```
Comment (addaleax) [inline source/export-to-language.txt:21]:
```suggestion
- Node.js
```

**Review by addaleax — APPROVED** [review]
